### PR TITLE
executor, expression: fix a panic of MAX/MIN

### PR DESCRIPTION
**Pull request by zz-jason**
For UDAF `max`/`min` whose parameter is a scala function with `float` as its return type, the result of `max`/`min` should be stored and returned as a `float64` value.

Let's say we have the following table:
```sql
DROP TABLE IF EXISTS T;
CREATE TABLE T(A VARCHAR(10), B VARCHAR(10), C FLOAT);
INSERT INTO T VALUES('0', "val_b", 12.191);
```

Before this PR, the following sql will cause a **"index out of range"** panic in tidb-server:
```sql
SELECT MAX(CASE B WHEN 'val_b'  THEN C ELSE 0 END) val_b FROM T WHERE cast(A as signed) = 0 GROUP BY a;
```

**NOTE**: the return type of `CASE B WHEN 'val_b'  THEN C ELSE 0 END` is `float32`

This PR fixes this panic.

### CODE REVIEW

**Comment by zz-jason** [comment]
@tiancaiamao @winoros @XuHuaiyu PTAL

File: executor/aggregate_test.go [hunk 1]
```diff
@@ -539,3 +539,17 @@ func (s *testSuite) TestIssue5663(c *C) {
 	tk.MustQuery("select group_concat(i) from t1 where i > 1;").Check(testkit.Rows("2,3"))
 	tk.MustQuery("select group_concat(i) from t1 where i > 1;").Check(testkit.Rows("2,3"))
 }
+
+func (s *testSuite) TestMaxMinFloatScalaFunc(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	plan.GlobalPlanCache = kvcache.NewShardedLRUCache(2, 1)
```
Comment (shenli) [inline executor/aggregate_test.go]: Is this issue related with plan-cache?

Comment (zz-jason) [reply executor/aggregate_test.go]: No, this is an irrelevant change, already reverted.

**Comment by shenli** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
typo: s/ scala/ scalar

File: expression/aggregation/descriptor.go [hunk 2]
```diff
@@ -210,6 +210,15 @@ func (a *AggFuncDesc) typeInfer4GroupConcat(ctx sessionctx.Context) {
 }
 
 func (a *AggFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
+	_, argIsScalaFunc := a.Args[0].(*expression.ScalarFunction)
+	if argIsScalaFunc && a.Args[0].GetType().Tp == mysql.TypeFloat {
+		// For scala functon, the result of "float32" is set to the "float64"
```
Comment (lamxTyler) [inline expression/aggregation/descriptor.go]: typo `functon`.

Comment (zhexuany) [reply expression/aggregation/descriptor.go]: typo scalar not scala.

**Comment by zhexuany** [comment]
@XuHuaiyu PTAL
